src/expressions/if-expr.md

```diff
@@ -126,30 +104,43 @@ if let E::X(n) | E::Y(n) = v {
 ```
 
 The expression cannot be a [lazy boolean operator expression][_LazyBooleanOperatorExpression_].
-Use of a lazy boolean operator is ambiguous with a planned feature change of the language (the implementation of if-let chains - see [eRFC 2947][_eRFCIfLetChain_]).
-When lazy boolean operator expression is desired, this can be achieved by using parenthesis as below:
 
-<!-- ignore: psuedo code -->
-```rust,ignore
-// Before...
-if let PAT = EXPR && EXPR { .. }
+## Chains of expressions
+
+It is possible to chain multiple expressions into a single statement to avoid nested declarations.
 
-// After...
-if let PAT = ( EXPR && EXPR ) { .. }
+```rust
+fn single() {
+    let outer_opt = Some(Some(1i32));
+
+    if let Some(inner_opt) = outer_opt
+        && let Some(number) = inner_opt
+        && number == 1
+    {
+        println!("Peek a boo");
+    }
+}
```

Comment:
This example isn't super clear to me. Can it perhaps be broken up in two and explain what it is illustrating?  Maybe something like this:

~~~suggestion

The following is an example of chaining multiple expressions, mixing `let` bindings and boolean expressions, and with expressions able to reference pattern bindings from previous expressions:

```rust
let outer_opt = Some(Some(1i32));

if let Some(inner_opt) = outer_opt
    && let Some(number) = inner_opt
    && number == 1
{
    println!("Peek a boo");
}
```

The above is equivalent to the following without using expression chains:

```rust
~~~

Reply:
Done